editor/src/components/canvas/canvas-strategies/look-for-applicable-parent.tsx

```diff
@@ -0,0 +1,298 @@
+      show: 'visible-only-while-active',
+    },
+    {
+      control: ParentBounds,
+      key: 'parent-bounds-control',
+      show: 'visible-only-while-active',
+    },
+  ],
+  isApplicable: (canvasState, interactionSession, metadata, allElementProps) => {
+    if (interactionSession == null) {
+      return false
+    }
+
+    const strategies = isApplicableInner(canvasState, interactionSession, metadata, allElementProps)
+    return strategies != null
+  },
+
+  fitness: (canvasState, interactionSession, strategyState) => {
+    return lookForApplicableParentStrategy.isApplicable(
+      canvasState,
+      interactionSession,
+      strategyState.startingMetadata,
+      strategyState.startingAllElementProps,
+    )
+      ? 1
+      : 0
+  },
+
+  apply: (canvasState, interactionSession, strategyState) => {
+    const result = isApplicableInner(
+      canvasState,
+      interactionSession,
+      strategyState.startingMetadata,
+      strategyState.startingAllElementProps,
+    )
+
+    if (result == null) {
+      return emptyStrategyApplicationResult
+    }
+
+    const { strategies, effectiveTarget, originalTarget } = result
+    if (strategies.length < 1) {
+      return emptyStrategyApplicationResult
+    }
+
+    const patchedCanvasState = patchCanvasStateInteractionTargetPath(canvasState, effectiveTarget)
+
+    const chosenStrategy = calcFittestStrategy(
+      result.strategies,
+      patchedCanvasState,
+      interactionSession,
+      strategyState,
+    )
+
+    const chosenStrategyApplicationResult = chosenStrategy.apply(
+      patchedCanvasState,
+      interactionSession,
+      strategyState,
+    )
+
+    return strategyApplicationResult(
+      [
+        ...chosenStrategyApplicationResult.commands,
+        highlightElementsCommand(effectiveTarget),
+        updateSelectedViews('mid-interaction', originalTarget),
+        setCursorCommand('mid-interaction', CSSCursor.MovingMagic),
+      ],
+      chosenStrategyApplicationResult.customStatePatch,
+      chosenStrategyApplicationResult.status,
+    )
+  },
+}
+
+function* elementAncestry(path: ElementPath) {
+  let currentParentPath = path
+  const componentsInSubtree: Array<ElementPath> = [path]
+  while (!EP.isStoryboardChild(currentParentPath)) {
+    yield {
+      root: currentParentPath,
+      componentsInSubtree,
+    }
+    const parent = EP.parentPath(currentParentPath)
+    componentsInSubtree.push(parent)
+    currentParentPath = parent
+  }
+}
+
+function isSingletonAbsoluteMove(strategies: Array<CanvasStrategy>): boolean {
+  return strategies.length === 1 && strategies[0].id === 'CONVERT_TO_ABSOLUTE_AND_MOVE_STRATEGY'
+}
+
+function patchCanvasStateInteractionTargetPath(
+  canvasState: InteractionCanvasState,
+  path: ElementPath[],
+): InteractionCanvasState {
+  return {
+    ...canvasState,
+    interactionTarget: {
+      type: 'TARGET_PATHS',
+      elements: path,
+    },
+  }
+}
+
+function isApplicableInner(
```

Comment:
I think this could use a better name, if I understand correctly, this is the function that returns a potential ancestor's applicable strategy. the name implies to me that it only returns isApplicable ~boolean